arch/eventloop_posix_udp.c

```diff
@@ -1075,21 +1088,13 @@ registerSocketAndDestinationForSend(size_t paramsSize, const UA_KeyValuePair *pa
                      "UDP\t| Configuring send multicast failed");
         return UA_STATUSCODE_BADINTERNALERROR;
     }
-    /* Non-blocking connect */
-    error = UA_connect(newSock, info->ai_addr, info->ai_addrlen);
-    if(error != 0 &&
-       UA_ERRNO != UA_INPROGRESS &&
-       UA_ERRNO != UA_WOULDBLOCK) {
-        UA_LOG_SOCKET_ERRNO_WRAP(
-            UA_LOG_WARNING(logger, UA_LOGCATEGORY_NETWORK,
-                           "UDP\t| Connecting the socket to %s failed (%s)",
-                           hostname, errno_str));
-        UA_close(newSock);
-        return UA_STATUSCODE_BADDISCONNECT;
-    }
+    memcpy(&ufd->sendAddr, info->ai_addr, sizeof(struct sockaddr));
```

Comment:
Use `info->ai_addrlen` instead of `sizeof(struct sockaddr)`.
The current code might not memcpy over everything.
Is it possible that struct sockaddr ist just a "basetype" with additional stuff depending on the actual protocol?